packages/gatsby-adapter-netlify/src/__tests__/pretty-urls.ts

```diff
@@ -0,0 +1,217 @@
+
+      ensureStaticAssetPath(`public/[param]/index.html`, `/foo/:param/`)
+
+      await fileMovingDone()
+
+      expect(vol.existsSync(`public/foo/[param]/index.html`)).toEqual(true)
+      expect(vol.existsSync(`public/[param]/index.html`)).toEqual(false)
+    })
+
+    it(`handles dynamic named wildcard paths syntax (is not using reserved characters for file paths)`, async () => {
+      vol.fromJSON(
+        { "public/[...wildcard]/index.html": `*wildcard` },
+        process.cwd()
+      )
+
+      const { ensureStaticAssetPath, fileMovingDone } =
+        createStaticAssetsPathHandler()
+
+      expect(vol.existsSync(`public/[...wildcard]/index.html`)).toEqual(true)
+      expect(vol.existsSync(`public/foo/[...].html`)).toEqual(false)
+
+      ensureStaticAssetPath(`public/[...wildcard]/index.html`, `/foo/*`)
+
+      await fileMovingDone()
+
+      expect(vol.existsSync(`public/foo/[...].html`)).toEqual(true)
+      expect(vol.existsSync(`public/[...wildcard]/index.html`)).toEqual(false)
+    })
+
+    it(`handles dynamic unnamed wildcard paths syntax (is not using reserved characters for file paths)`, async () => {
+      vol.fromJSON({ "public/[...]/index.html": `*` }, process.cwd())
+
+      const { ensureStaticAssetPath, fileMovingDone } =
+        createStaticAssetsPathHandler()
+
+      expect(vol.existsSync(`public/[...]/index.html`)).toEqual(true)
+      expect(vol.existsSync(`public/foo/[...].html`)).toEqual(false)
+
+      ensureStaticAssetPath(`public/[...]/index.html`, `/foo/*`)
+
+      await fileMovingDone()
+
+      expect(vol.existsSync(`public/foo/[...].html`)).toEqual(true)
+      expect(vol.existsSync(`public/[...]/index.html`)).toEqual(false)
+    })
+
+    it(`keeps 404.html in root`, async () => {
+      vol.fromJSON({ "public/404.html": `404` }, process.cwd())
+
+      const { ensureStaticAssetPath, fileMovingDone } =
+        createStaticAssetsPathHandler()
+
+      expect(vol.existsSync(`public/404.html`)).toEqual(true)
+      expect(vol.existsSync(`public/foo/404.html`)).toEqual(false)
+
+      ensureStaticAssetPath(`public/404.html`, `/foo/404.html`)
+
+      await fileMovingDone()
+
+      expect(vol.existsSync(`public/404.html`)).toEqual(true)
+      // 404 page is coped so it exists in both locations
+      expect(vol.existsSync(`public/foo/404.html`)).toEqual(true)
+    })
+
+    it(`keeps 500.html in root`, async () => {
+      vol.fromJSON({ "public/500.html": `500` }, process.cwd())
+
+      const { ensureStaticAssetPath, fileMovingDone } =
+        createStaticAssetsPathHandler()
+
+      expect(vol.existsSync(`public/500.html`)).toEqual(true)
+      expect(vol.existsSync(`public/foo/500.html`)).toEqual(false)
+
+      ensureStaticAssetPath(`public/500.html`, `/foo/500.html`)
+
+      await fileMovingDone()
+
+      expect(vol.existsSync(`public/500.html`)).toEqual(true)
+      // 404 page is coped so it exists in both locations
```

Comment:
comment should be `500` page?

Reply:
fixed in https://github.com/gatsbyjs/gatsby/pull/38666/commits/28856c21b8eddafcfdb902ba020192bc902fabda